File: backend/src/main/java/com/wonlinestore/backend/entity/ClientOrder.java
package com.wonlinestore.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class ClientOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false)
    private Date date;
    @ManyToOne
    private Client client;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    private List<CartItem> cartItems;
    private BigDecimal lineTotal;


}

File: frontend/src/types/ClientOrder.ts
import {Client} from "./Client.ts";
import {CartItem} from "./CartItem.ts";

export class ClientOrder {
id: number;
date: string;
client: Client;
cartItems: CartItem[];
lineTotal: number;

    constructor(id: number, date: string, client: Client, cartItems: CartItem[], lineTotal: number) {
        this.id = id;
        this.date = date;
        this.client = client;
        this.cartItems = cartItems;
        this.lineTotal = lineTotal;
    }
}